components/dashboards/org.wso2.carbon.dashboards.api/src/main/java/org/wso2/carbon/dashboards/api/internal/DashboardRestApi.java

```diff
@@ -120,11 +129,16 @@ public Response get() {
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
     @Path("/{id}")
-    public Response get(@PathParam("id") String id) {
+    public Response get(@PathParam("id") String id, @Context Request request) {
         try {
-            return dashboardDataProvider.get(id)
-                    .map(metadata -> Response.ok().entity(metadata).build())
+            return dashboardDataProvider.getDashboardByUser(request.getProperty(InterceptorConstants
+                            .PROPERTY_USERNAME).toString(),
+                    id, request.getHeader("X-Dashboard-Origin-Component")).map(metadata ->
+                    Response.ok().entity(metadata).build())
                     .orElse(Response.status(NOT_FOUND).entity("Cannot find a dashboard for ID '" + id + "'.").build());
+        } catch (DashboardUnauthorizedException e) {
+            return Response.status(UNAUTHORIZED).entity("You are not authorized to access the dashboard with ID" +
```

Comment:
In this error message `You are` part is incorrect. It assumes that there is always a human user who invoked the REST API, which might be incorrect some times. So, shall we change the wording of this error message e.g. `Insufficient permissions to retrieve dashboard ' + id + '.`

Apply for other places as well.